File: 1001-n-repeated-element-in-size-2n-array/1001-n-repeated-element-in-size-2n-array.c
int repeatedNTimes(int* nums, int numsSize) {
    int max = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] > max) {
            max = nums[i];
        }
    }
    int* freq = calloc(max + 1, sizeof(int));
    for (int i = 0; i < numsSize; i++) {
        freq[nums[i]]++;
        if (freq[nums[i]] > 1) {
            return nums[i];
        }
    }
    return -1;
}

File: 1001-n-repeated-element-in-size-2n-array/1001-n-repeated-element-in-size-2n-array.java
class Solution {
    public int repeatedNTimes(int[] nums) {
        HashMap<Integer,Integer> freq = new HashMap<>();
        for(int num:nums){
            freq.put(num,freq.getOrDefault(num,0)+1);
            if(freq.get(num)>1){
                return num;
            }
        }
        return -1;
    }
}